test_DslashBSM2.c

```diff
@@ -380,10 +380,12 @@ if( strcmp(scalar_input_filename, "create_random_scalarfield") == 0 ) {
 		printf("\n# square norm of the source: ||w||^2 = %e\n\n", squarenorm);
 		fflush(stdout);
 	}
-
+//initialize BSM2f operator
+ init_D_psi_BSM2f();
```

Comment:
out of interest, what is the overhead of initialising and freeing this one per gauge configuration?